pkg/statistics/histogram.go

```diff
@@ -1542,40 +1608,48 @@ func MergePartitionHist2GlobalHist(sc *stmtctx.StatementContext, hists []*Histog
 	}
 	if r > 0 {
 		bucketSum := int64(0)
-		for _, b := range buckets[:r] {
+		leftMost := buckets[0].lower
+		for i, b := range buckets[:r] {
 			bucketSum += b.Count
+			if i == 0 {
+				continue
+			}
+			res, err := leftMost.Compare(sc.TypeCtx(), b.lower, collate.GetBinaryCollator())
+			if err != nil {
+				return nil, err
+			}
+			if res > 0 {
+				leftMost = b.lower
+			}
 		}
 
 		if len(globalBuckets) > 0 && bucketSum < gBucketCountThreshold { // merge them into the previous global bucket
```

Comment:
I think this case will also be incorrect if "overlap and cut" happens. Because in this case, the `buckets` has already been modified and is incomplete.

Reply:
The fix is too long to introduce. I just removed this if block.